src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+    
+
+    wave_dim, y_dim, x_dim  = np.shape(red_flux_cube)
+    wave_dim = len(wl)
+
+    flux_cube = np.zeros((wave_dim, y_dim, x_dim))
+    fluxvar_cube = np.zeros((wave_dim, y_dim, x_dim))
+
+
+
+    # Run over every point (i,j) in the spatial plane
+    for i in range(y_dim):
+        for j in range(x_dim):
+
+            red_flux =  red_flux_cube[:, i, j]
+            red_fluxvar =  red_fluxvar_cube[:, i, j]
+
+            blue_flux =  blue_flux_cube[:, i, j]
+            blue_fluxvar =  blue_fluxvar_cube[:, i, j]
+
+            flux_R = np.array(AR * red_flux).ravel()
+            flux_B = np.array(AB * blue_flux).ravel()
+
+            # Two sparse diagonal matrices
+            diag_R = sp.dia_matrix(([red_fluxvar], [0]), shape=[redNAXIS1, redNAXIS1])
+            diag_B = sp.dia_matrix(([blue_fluxvar], [0]), shape=[blueNAXIS1, blueNAXIS1])
+
+            fluxvar_B = np.array((AB * diag_B * AB.T).sum(axis=1)).ravel()
+            fluxvar_R = np.array((AR * diag_R * AR.T).sum(axis=1)).ravel()
+
+            buffer = 10.0
+
+            # Should check for wavelength overlap
+
+            blueOnly = np.where(wl <= red_minWL + buffer)
+            overlap = np.where((wl < blue_maxWL - buffer) & (wl > red_minWL + buffer))
+            redOnly = np.where(wl >= blue_maxWL - buffer)
+
+            if red_minWL > blue_maxWL:
+                
+                return None, None
+
+            else:
+                # Average the two taking into account the buffer region and weighting
+                flux = np.zeros(len(flux_B), float)
+                fluxVar = np.zeros(len(flux_B), float)
+
+                flux[blueOnly] = flux_B[blueOnly]
+                fluxVar[blueOnly] = fluxvar_B[blueOnly]
+
+                fluxVar[overlap] = 1.0 / (1.0 / fluxvar_B[overlap] + 1.0 / fluxvar_R[overlap])
+                flux[overlap] = (
+                    flux_B[overlap] / fluxvar_B[overlap] + flux_R[overlap] / fluxvar_R[overlap]
+                ) * fluxVar[overlap]
+
+                flux[redOnly] = flux_R[redOnly]
+                fluxVar[redOnly] = fluxvar_R[redOnly]
+                flux_cube[:,i,j] = flux
+                fluxvar_cube[:,i,j] = fluxVar
+
+
+    # Reshape the cube to the corresponding shape
+    wave_dim = len(flux)
+    flux_cube = flux_cube.reshape(wave_dim, y_dim, x_dim)
+    fluxvar_cube = fluxvar_cube.reshape(wave_dim, y_dim, x_dim)
+
+    return flux_cube, fluxvar_cube
+
+
+
+def splice_cubes(blue_path, red_path, output):
+    """
+    The main routine
+    """
+
+    red_header = fits.getheader(red_path, 0)
+    blue_header = fits.getheader(blue_path, 0)
+
+
+    # Join the cubes
+    print("Splicing cubes")
+    flux, fluxVar = join_cubes(blue_path, red_path)
+
+    if len(flux) == 1 :
+        print("No spectral overlap")
+
+    else:
+
+        # Write out the results
+        # Use the header in red arm to start with
+        # Add additional blue CCD keywords as required - pending
```

Comment:
Does this comment need an update? It ends with `pending`. If you'd like to track upcoming changes, you could make a github issue.